lib/fetching/content-services/crowd-tangle-content-service.js

```diff
@@ -184,7 +185,20 @@ CrowdTangleContentService.prototype._parse = function(data, options) {
 	if (this.crowdtangle_lists[data.account.id]) {
 		metadata.ct_tag = this.crowdtangle_lists[data.account.id];
 	} else if (options.requestType == "SAVED_SEARCH") {
-		metadata.ct_tag = "Saved Search";
+		metadata.ct_tag = []
+		for(var saved_search_id in this.crowdtangle_saved_searches){
+			var saved_search = this.crowdtangle_saved_searches[saved_search_id];
+			var regex = new RegExp(saved_search.words.join("|"), 'gi');
+			if(content.search(regex) >= 0 || 
+				(metadata.caption !== null && metadata.caption.search(regex) >=0) || 
+				(metadata.description !== null && metadata.description.search(regex) >= 0) ||
+				(metadata.imageText !== null && metadata.imageText.search(regex) >= 0)) {
```

Comment:
Tip: Instead of `if not null && string.search(regex)` you could simplify to `regex.test(string)` if you want to. No null check needed that way.